viper/parser/expr.py

```diff
@@ -301,6 +306,9 @@ def arithmetic(self):
             return LLLnode.from_list(['clamp', ['mload', MemoryPositions.MINNUM], o, ['mload', MemoryPositions.MAXNUM]], typ=o.typ, pos=getpos(self.expr))
         elif o.typ.typ == 'decimal':
             return LLLnode.from_list(['clamp', ['mload', MemoryPositions.MINDECIMAL], o, ['mload', MemoryPositions.MAXDECIMAL]], typ=o.typ, pos=getpos(self.expr))
+        elif o.typ.typ == 'num256':
+            # No clamping required, takes up full register
```

Comment:
So this is where it gets funky with taking up the full 32byte word. We still actually have to protect against overflow hey, clamping makes that easy to fail with `num` type. With this scenario we'd have to find each operator '* ^ +' and make sure there is no overflow. Just something to keep in mind ;)

Reply:
Oh you mean like specialty code? (like safemath) Yeah that makes sense. Weren't we looking to implement safemath more generally for all types? I remember @DavidKnott talking about this.

Reply:
With that approach, much later down the line we can do a model checker on the code and see if it can violate the clamp before we add it as an optimization step, for things like tokens (unless it's mintable, and even then we only need to ensure the total supply never overflows) since most tokens can never overflow.

Way future thoughts though.